File: src/types/input_parameter.ts
import { InputValueResponse } from "./input_value";

export type CreateInputParameterRequest = {
    problem_id: number;
}

export type UpdateInputParameterRequest = {
    name: string;
    start: number;
    end: number;
}

export type InputParameterResponse = {
    id: number;
    problem_id: number;
    name: string;
    start: number;
    end: number;
    input_values: Array<InputValueResponse>;
}

File: src-tauri/src/domain/entities/input_parameter.rs
use serde::{Deserialize, Serialize};

use crate::domain::entities::input_value::InputValue;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct InputParameter {
    pub id: i64,
    pub problem_id: i64,
    pub name: String,
    pub start: f32,
    pub end: f32,
    pub input_values: Vec<InputValue>,
}
